Add article listing prototype

dromo77 opened the pull request:
## Overview

This PR adds a new article listing prototype. It includes recent articles, featured articles, and additional topics.

## Screenshots

### Large Screens

![Screen Shot 2021-07-08 at 11 36 14 AM](https://user-images.githubusercontent.com/42841342/124985126-ea3cdd80-dfee-11eb-9d1b-6d5aa254ff9a.png)
![Screen Shot 2021-07-08 at 11 36 25 AM](https://user-images.githubusercontent.com/42841342/124985137-ec9f3780-dfee-11eb-83b3-3cb56128bced.png)
![Screen Shot 2021-07-08 at 11 36 34 AM](https://user-images.githubusercontent.com/42841342/124985149-ef9a2800-dfee-11eb-98ec-d7033bdb642f.png)
![Screen Shot 2021-07-09 at 11 47 23 AM](https://user-images.githubusercontent.com/42841342/125123500-97752b80-e0ab-11eb-804b-cda016bc25d7.png)

### Small Screens
![Screen Shot 2021-07-08 at 11 38 26 AM](https://user-images.githubusercontent.com/42841342/124985243-093b6f80-dfef-11eb-8427-00e3dc6abca7.png)
![Screen Shot 2021-07-09 at 11 48 04 AM](https://user-images.githubusercontent.com/42841342/125123543-a5c34780-e0ab-11eb-8dff-31bc520d2c2a.png)
![Screen Shot 2021-07-09 at 11 47 36 AM](https://user-images.githubusercontent.com/42841342/125123558-a956ce80-e0ab-11eb-8aa2-bf7020611461.png)

## Testing

- [Deploy Preview](https://deploy-preview-1372--cloudfour-patterns.netlify.app/?path=/story/prototypes-article-listing--example)

tylersticka commented on src/prototypes/article-listing/example/example.twig:
```diff
@@ -0,0 +1,145 @@
+          {% embed '@cloudfour/components/button/button.twig' %}
+            {% block content %}
+              {% include '@cloudfour/components/icon/icon.twig' with {
+                name: 'search'
```
Make sure you rebase or merge in the latest `v-next`, as some icon names (such as this one) have changed

tylersticka commented on src/prototypes/article-listing/example/example.scss:
```diff
@@ -0,0 +1,110 @@
+
+  .section-title {
+    font-weight: 300;
+    font-size: ms.step(5);
```
Did you consider using one of our `fluid.*` mixins to set a `font-size` that changes between a min and max depending on viewport? These currently feel a little too large on small screens and this might remedy that. (See our heading component CSS for an example.)

tylersticka commented on src/prototypes/article-listing/example/example.twig:
```diff
@@ -0,0 +1,145 @@
+          {% block content %}
+            <div class="title-content">
+              <h2 class="section-title">{{topic.title}}</h2>
+              <div>
```
Can these topic buttons move to _after_ the article listings on small screens rather than after the headings?

tylersticka commented on src/prototypes/article-listing/example/example.twig:
```diff
@@ -0,0 +1,145 @@
+      <div class="topics">
+        <h2 class="section-title">More Topics</h2>
+        <div class="search-content">
+          <label for="search-input" class="search-label">Search</label>
```
Something is feeling unresolved about this currently.

Should the button and form field be joined? Can the icon be sized a little more nicely? Is the label necessary, and in the right place? 🤔

dromo77 commented:
Hey @tylersticka! I made a few changes to the prototype:

- Merged in the latest from v-next
- Adjusted the heading size so it's fluid
- Moved the "topic" buttons beneath the cards on small screens
- Visually joined the search input and button
- Removed the search bar label

tylersticka commented:
@dromo77 Thanks for making those changes!

When I try to view this prototype in the deploy preview, I see an error related to outdated icon names (`close` and `search`).

Also, I forgot to mention last time that we need to figure out where pagination should go or if there needs to be an "all articles" view or something like that. How might someone continue browsing articles regardless of topic?

dromo77 commented:
@tylersticka It's weird that I wasn't getting the same error, but I updated the icon name. Let me know if you're still having problems.

I added pagination beneath the first section of recent articles. I'd like to know what you think. I'm open to exploring some more ideas like an "all articles" view.

tylersticka commented:
> Let me know if you're still having problems.

It looks to be fixed! And FYI, I saw this on the deploy preview, which you can see by expanding the checks that have passed at the bottom of a PR and clicking the "Details" link next to `deploy/netlify`.
